data-pipeline/epimodel/gleam/scenario.py

```diff
@@ -44,40 +45,87 @@ def __init__(self, rds: RegionDataset, foretold_token=None, progress_bar=True):
                 .apply(self._get_region)
                 .apply(lambda reg: reg.Code),
-                "Infectious": est["Infectious_mean"].astype("float"),
+                "Infectious": est["Infectious"].astype("float"),
+                "Exposed": est["Exposed"].astype("float"),
             }
-        ).set_index("Region")["Infectious"]
+        ).set_index("Region")
+
+        infectious_distributed = compartments_df["Infectious"]
+        exposed_distributed = compartments_df["Exposed"]
 
         # modifies series in-place, adding to the index
-        algorithms.distribute_down_with_population(infectious, self.rds)
+        algorithms.distribute_down_with_population(infectious_distributed, self.rds)
+        algorithms.distribute_down_with_population(exposed_distributed, self.rds)
 
-        df = pd.DataFrame(infectious).reset_index()
+        df = pd.DataFrame([infectious_distributed, exposed_distributed]).T
+        df.index = df.index.set_names(["Region"])
+        df = df.reset_index()
         df["Region"] = df["Region"].apply(self._get_region)
         is_gleam_basin = df["Region"].apply(
             lambda region: region.Level == Level.gleam_basin
         )
 
-        return df[is_gleam_basin][self.ESTIMATE_FIELDS].dropna()
+        return df.loc[is_gleam_basin, :].dropna()
+
+    def parse_country_scenarios_df(self, raw_estimates: pd.DataFrame):
+        est = raw_estimates.replace({"": None, "#N/A": None}).dropna(subset=["Name"])
+
+        scenario_columns = [
+            scenario_col
+            for scenario_col in est.columns
+            if scenario_col.startswith("scenario")
+        ]
+        scenarios = {
+            "Region": est["Name"].apply(self._get_region).apply(lambda reg: reg.Code),
+        }
+        scenarios.update(
+            {
+                scenario_col: est[scenario_col].astype("float")
+                for scenario_col in scenario_columns
+            }
+        )
+        df = pd.DataFrame(scenarios).set_index("Region")
+
+        algorithms.assign_down_with_population(df, self.rds)
+
+        df.index = df.index.set_names(["Region"])
+        df = df.reset_index()
+        df["Region"] = df["Region"].apply(self._get_region)
+        is_gleam_basin = df["Region"].apply(
+            lambda region: region.Level == Level.gleam_basin
+        )
+
+        return df.loc[is_gleam_basin, ["Region"] + scenario_columns].dropna()
 
     def parse_parameters_df(self, raw_parameters: pd.DataFrame):
         df = raw_parameters.replace({"": None})[self.PARAMETER_FIELDS].dropna(
             subset=["Parameter"]
         )
+
         df["Start date"] = pd.to_datetime(df["Start date"])
         df["End date"] = pd.to_datetime(df["End date"])
+
+        # Setting the time range of parameters with nan dates to the time range of the run
+        df.loc[df["Start date"].isna(), "Start date"] = df.loc[
+            df["Parameter"] == "run dates", "Start date"
+        ].item()
```

Comment:
i would extract
```
df.loc[
            df["Parameter"] == "run dates", "Start date"
        ].item()
```
into a variable since the assignment is used twice, but more importantly its a fairly complicated part of code. When i was reading it it was not immediately obvious that start date is a scalar, at first it looked like a vector (series).